plugins/modules/nd_service.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Sabari Jaganathan (@sajagana) <sajagana@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: nd_service
+short_description: Manages Service Package on Nexus Dashboard.
+description:
+- Manages Service Package of the Nexus Dashboard.
+author:
+- Sabari Jaganathan (@sajagana)
+options:
+  import_url:
+    description:
+    - The remote location of the Service Package.
+    aliases: [ url ]
+    type: str
+  import_id:
+    description:
+    - The ID of the imported Service Package.
+    aliases: [ id ]
+    type: str
+  state:
+    description:
+    - Use C(present) for importing a Service Package. The C(present) state is not idempotent.
+    - Use C(query) for listing all Service Packages.
+    - Use C(absent) for deleting a Service Package.
+    type: str
+    choices: [ present, query, absent ]
+    default: present
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Import a service package
+  cisco.nd.nd_service:
+    import_url: "https://nd_service.cisco.com/cisco-terraform-v0.1.16.aci"
+    state: present
+  register: import_result
+
+- name: Query a service package with import_id
+  cisco.nd.nd_service:
+    import_id: "{{ import_result.current.metadata.id }}"
+    state: query
+  register: query_result_import_id
+  until:
+    - query_result_import_id.current is defined
+    - query_result_import_id.current != {}
+    - query_result_import_id.current.status.downloadPercentage == 100
+  retries: 5
+  delay: 5
+
+- name: Remove a service package with import_id
+  cisco.nd.nd_service:
+    import_id: "{{ query_result_import_id.current.metadata.id }}"
+    state: absent
+
+# When the import_url has exactly one match
+- name: Remove a service package with import_url
+  cisco.nd.nd_service:
+    import_url: "http://173.36.219.254/cisco-terraform-v0.1.15.aci"
+    state: absent
+
+- name: Query all service packages with import_url
+  cisco.nd.nd_service:
+    import_url: "https://nd_service.cisco.com/cisco-terraform-v0.1.16.aci"
+    state: query
+  register: query_reseult_import_url
+
+# When the query_reseult_import_url has more than one match
+- name: Remove all query_reseult_import_url result service packages with import_id
```

Comment:
Remove all service packages with import_id when the query_result_import_url has more than one match